code/game/turfs/turf_changing.dm

```diff
@@ -17,8 +23,8 @@
 	// This makes sure that turfs are not changed to space when one side is part of a zone
 	if(N == /turf/space)
 		var/turf/below = GetBelow(src)
-		if(istype(below) && (air_master.has_valid_zone(below) || air_master.has_valid_zone(src)))
-			N = /turf/simulated/open
+		if(istype(below) && !istype(below,/turf/space))
+			N = below.density ? /turf/simulated/floor/airless : /turf/simulated/open
```

Comment:
This should probably get the filled turf type from the area instead of always using a floor.